### PR TITLE
Update //build to m114.0.5735.331.

### PR DESCRIPTION
**Pull request by ghost**
b/313662336

### CODE REVIEW

**Comment by andrewsavage1** [comment]
Could you please retry the failed tests to ensure they're flakes? Also please test on device and internally

**Review by andrewsavage1 — CHANGES_REQUESTED** [review]
Why are so many things outside of //build changing in this PR? The update should only affect that directory and possibly direct references to it, but there's a lot of other things going on here

**Review by andrewsavage1 — APPROVED** [review]

File: build/config/ui.gni [hunk 1]
```diff
@@ -20,14 +20,9 @@
...
+import("//build/config/ozone.gni")
 
 declare_args() {
-  # Indicates if Ozone is enabled. Ozone is a low-level library layer for Linux
-  # that does not require X11.
-  use_ozone =
-      is_chromeos || (is_chromecast && !is_android) || is_fuchsia ||
-      is_linux && !is_starboard
-
   # Indicates if the UI toolkit depends on X11.
```
Comment (dahlstrom-g) [inline build/config/ui.gni:26]: Lines 26–30 aren't in the source revision:
https://chromium.googlesource.com/chromium/src/+/114.0.5735.331/build/config/ui.gni#25

File: build/config/ui.gni [hunk 2]
```diff
@@ -42,18 +37,16 @@ declare_args() {
   # True means the UI is built using the "views" framework.
   toolkit_views = is_mac || is_win || is_linux || is_chromeos || is_fuchsia
 
-  # TODO(crbug.com/1171629): Remove is_chromeos_lacros.
-  # Whether we should use glib, a low level C utility library.
-  use_glib = (is_linux || is_chromeos_lacros) && !is_chromecast && !is_starboard
+  use_glib =
+      is_linux && !is_castos && !is_starboard &&
```
Comment (dahlstrom-g) [inline build/config/ui.gni:41]: ditto about inserting a line instead of appending `!is_starboard &&`

File: build/lacros/lacros_resource_sizes.pydeps
Comment (dahlstrom-g) [inline build/lacros/lacros_resource_sizes.pydeps:1]: Line 3 isn't in the source revision:
https://chromium.googlesource.com/chromium/src/+/114.0.5735.331/build/lacros/lacros_resource_sizes.pydeps#3

File: build/lacros/test_runner.py [hunk 3]
```diff
@@ -458,7 +809,39 @@ def Main():
...
+  # This is for version skew testing. The current CI/CQ builder builds
+  # an ash chrome and pass it using --ash-chrome-path. In order to use the same
+  # builder for version skew testing, we use a new argument to override
+  # the ash chrome.
+  test_parser.add_argument(
+      '--ash-chrome-path-override',
+      type=str,
+      help='The same as --ash-chrome-path. But this will override '
+      '--ash-chrome-path or --ash-chrome-version if any of these '
+      'arguments exist.')
```
Comment (dahlstrom-g) [inline build/lacros/test_runner.py:832]: Lines 823–832 appear to duplicate lines 846–855.

File: build/toolchain/apple/toolchain.gni [hunk 4]
```diff
@@ -89,6 +129,34 @@ template("apple_toolchain") {
...
+    #
+    # For this reason, "ios" needs to override `is_component_build` as its
+    # default value depends on `current_os`. Use the overridden value if it
+    # is set in `toolchain_args`.
+    if (defined(toolchain_args.is_component_build)) {
+      toolchain_is_component_build = toolchain_args.is_component_build
+    } else {
+      toolchain_is_component_build = is_component_build
+    }
+
```
Comment (dahlstrom-g) [inline build/toolchain/apple/toolchain.gni:159]: Lines 160–173 aren't in the source revision:
https://chromium.googlesource.com/chromium/src/+/114.0.5735.331/build/toolchain/apple/toolchain.gni#160

File: build/toolchain/gcc_toolchain.gni [hunk 5]
```diff
@@ -396,9 +430,9 @@ template("gcc_toolchain") {
       # now to add cflags for evergreen platforms but we haven't yet decided
       # whether cflags should be added here for all platforms.
       if (!is_starboard_toolchain && is_starboard && sb_is_modular) {
-        command = "$asm -MMD -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
+        command = "$asm $md -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{cflags}} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
       } else {
-      command = "$asm -MMD -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
+        command = "$asm $md -MF $depfile ${rebuild_string}{{defines}} {{include_dirs}} {{asmflags}}${extra_asmflags} -c {{source}} -o {{output}}"
```
Comment (dahlstrom-g) [inline build/toolchain/gcc_toolchain.gni:435]: ditto about indentation

File: build/toolchain/ios/compile_xcassets.py [hunk 6]
```diff
@@ -30,38 +30,20 @@
...
-    if match is not None:
-      return True
-  return False
+def FixAbsolutePathInLine(line, relative_paths):
+  """Fix absolute paths present in |line| to relative paths."""
+  absolute_path = line.split(':')[0]
+  relative_path = relative_paths.get(absolute_path, absolute_path)
+  if absolute_path == relative_path:
+    return line
+  return relative_path + line[len(absolute_path):]
```
Comment (dahlstrom-g) [inline build/toolchain/ios/compile_xcassets.py:46]: Lines 40–48 appear to duplicate lines 49–57.

File: build/android/pylib/gtest/gtest_test_instance_test.py [hunk 7]
```diff
@@ -276,10 +276,41 @@ def testParseGTestJSON_example(self):
...
+        "seconds_since_epoch": 1406662283.764424,
+        "num_failures_by_type": {
+          "SKIP": 1
+        }
+      }"""
+    actual = gtest_test_instance.ParseGTestJSON(raw_json)
+    self.assertEqual(1, len(actual))
+    self.assertEqual('mojom_tests.parse.ast_unittest.ASTTest.testNodeBase',
+                     actual[0].GetName())
+    self.assertEqual(base_test_result.ResultType.SKIP, actual[0].GetType())
```
Comment (dahlstrom-g) [inline build/android/pylib/gtest/gtest_test_instance_test.py:313]: Lines 283–313 appear to duplicate lines 314–344.

File: build/android/pylib/results/json_results_test.py [hunk 8]
```diff
@@ -236,22 +225,50 @@ def testGenerateJsonTestResultFormatDict_failedResult(self):
...
+    # Should only be set if the test fails.
+    self.assertNotIn('is_unexpected',
+                     results_dict['tests']['test']['package']['TestName'])
+
+    self.assertTrue('FAIL' not in results_dict['num_failures_by_type']
+                    or results_dict['num_failures_by_type']['FAIL'] == 0)
+    self.assertTrue('PASS' not in results_dict['num_failures_by_type']
+                    or results_dict['num_failures_by_type']['PASS'] == 0)
+    self.assertIn('SKIP', results_dict['num_failures_by_type'])
+    self.assertEqual(1, results_dict['num_failures_by_type']['SKIP'])
```
Comment (dahlstrom-g) [inline build/android/pylib/results/json_results_test.py:271]: Lines 244–271 appear to duplicate lines 272–299.

**Review by dahlstrom-g — COMMENTED** [review]
This file isn't in the source revision:
https://github.com/youtube/cobalt/blob/fd2b36a1/build/config/fuchsia/test/web_instance_host_capabilities.test-cmx
https://chromium.googlesource.com/chromium/src/+/114.0.5735.331/build/config/fuchsia/test

**Pull request merged**